layers/+lang/ruby/funcs.el

```diff
@@ -22,11 +22,55 @@
 
 
 
-;; backend
+;; macros
+
+(defmacro spacemacs|eval-for-enabled-ruby-mode (body)
+  "Evaluate BODY for the currently active ruby mode.
+
+All occurrences of `mode' in BODY are replaced with the actual mode name symbol.
+All occurrences of `hook' in BODY are replaced with the actual hook name symbol.
+All occurrences of `local-vars-hook' in BODY are replaced with the actual local
+vars hook name symbol.
+Note that `mode', `hook' and `local-vars-hook' need to be explicitly quoted
+except when they are themselves inside a macro."
+  (declare (debug (form)) (indent 0))
+  (if ruby-enable-enh-ruby-mode
+      ;; enh-ruby-mode
+      (progn
+        (let ((new 'enh-ruby-mode)
+              (old 'mode))
+          (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body))
+        (let ((new 'enh-ruby-mode-hook)
+              (old 'hook))
+          (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body))
+        (let ((new 'enh-ruby-mode-local-vars-hook)
+              (old 'local-vars-hook))
+          (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body)))
+    ;; ruby-mode
+    (let ((new 'ruby-mode)
+          (old 'mode))
+      (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body))
+    (let ((new 'ruby-mode-hook)
+          (old 'hook))
+      (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body))
+    (let ((new 'ruby-mode-local-vars-hook)
+          (old 'local-vars-hook))
+      (cl-nsubstitute-if new 'spacemacs//ruby-substitute-predicate body))))
```

Comment:
This part can be simplified.
First we let-bound a list of modes based on whether `ruby-enable-enh-ruby-mode` is t.
```elisp
(let* ((rubymode (if ruby-enable-enh-ruby-mode
                     "enh-ruby"
                  "ruby")
       (pairs (cl-loop for old in '("mode" "hook" "local-vars-hook")
                       for new = (s-join "-" (list rubymode old))
                       collect (cons old new)))
 ..)
```
Then we can simply
```elisp
  (dolist (pair pairs)
     (cl-nsubstitute-if (cdr pair) 'spacemacs//ruby-substitute-predicate (car pair)))
```

Reply:
The real question is, how efficient is `cl-nsubstitute-if`.

Reply:
Good question, I would assume it is the fastest as it is the destructive version `cl-substitute-if` but this is a wild guess 😅 
Going to update with your great implementation for now.

Reply:
Done